### PR TITLE
[CPU] Set cpu core id from runtime parameters to determine which cores can be used by this OV instance

### PR DESCRIPTION
**Pull request by a3213105**
For stand-alone OpenVINO inference, we can use numactl or taskset to determine which CPU cores can be used by the OV process.
However, when we run inference in an application, we need to isolate the inference cores from other common process cores to ensure that each component gets the best hardware configuration.
So, we add this new runtime parameters to control the CPUs that can be used by OpenVINO instance.

If we don't set this parameter, the behavior will be the same as the current status

### CODE REVIEW

**Comment by mlukasze** [comment]
hey @a3213105 thanks for the PR
could you please resolve conflicts before we will enable CI checks for this change?

**Comment by a3213105** [comment]
> hey @a3213105 thanks for the PR could you please resolve conflicts before we will enable CI checks for this change?

ok，i will check them

File: samples/cpp/benchmark_app/main.cpp [hunk 1]
```diff
@@ -512,12 +512,23 @@ int main(int argc, char* argv[]) {
                     }
                 }
             };
+
+            auto set_cpu_core_ids = [&] {
```
Comment (akuporos) [inline samples/cpp/benchmark_app/main.cpp:516]: Should it also be added to python benchmark?

File: samples/cpp/benchmark_app/benchmark_app.hpp [hunk 2]
```diff
@@ -316,6 +320,8 @@ DEFINE_uint64(nireq, 0, infer_requests_count_message);
 /// @brief Number of streams to use for inference on the CPU (also affects Hetero cases)
 DEFINE_string(nstreams, "", infer_num_streams_message);
 
+DEFINE_string(cpu_core_ids, "", infer_cpu_core_ids_message);
```
Comment (Wovchena) [inline samples/cpp/benchmark_app/benchmark_app.hpp:323]: Please extend https://github.com/openvinotoolkit/openvino/blob/master/tests/samples_tests/smoke_tests/test_benchmark_app.py to run the new flag and the list of cmd args in https://github.com/openvinotoolkit/openvino/blob/master/docs/articles_en/learn-openvino/openvino-samples/benchmark-tool.rst.

**Comment by wangleis** [comment]
@a3213105 Thanks for your contribution. 

This feature include two part. One is user interface part include new property. The other is feature execution part in CPU plugin and interface folders.

From design perspective for feature execution part, we just need feature to update cpu_mapping_table and proc_type_table based on input processors before get_num_streams(). Then we do not need much API change with "executor_id". Could you please try it?

File: src/inference/src/system_conf.cpp [hunk 3]
```diff
@@ -413,18 +555,48 @@ void reserve_available_cpus(const std::vector<std::vector<int>> streams_info_tab
                                                stream_processors,
                                                cpu_status);
 
-    OPENVINO_DEBUG << "[ threading ] stream_processors:";
+    // printf("[ threading ] cpu_mapping_table (%ld):\n", cpu._cpu_mapping_table.size());
```
Comment (wangleis) [inline src/inference/src/system_conf.cpp:558]: Please remove debug code in PR.

**Comment by a3213105** [comment]
> get_num_streams

yes, i will think about it for the minimal change to the original code.

**Comment by a3213105** [comment]
For the workloads we've encountered recently, the entire pipeline has always had several models combined
We can assume that there is a pipeline below:

First_model -->  　Parralle_model_1  　----->  Last_model
　　　　　　| 　　　　　　　　　　　^    
　　　　　　-->  Parralle_model_2 　 __|
            
On a 16C32T server, we can bind the OV stream and core with the following:

core=OV::Core()

config1={}
config1['CPU_CORE_IDS']='0-15'
first_model = core.compile_model(First_model, "CPU", config1)

config2={}
config2['CPU_CORE_IDS']='0-7'
parralle_model_1 = core.compile_model(Parralle_model_1, "CPU", config2)

config3={}
config3['CPU_CORE_IDS']='8-15'
parralle_model_2 = core.compile_model(Parralle_model_2, "CPU", config3)

config4={}
config4['CPU_CORE_IDS']='0-15'
last_model = core.compile_model(Last_model, "CPU", config4)


So, we need store the whole CPU info in CPU_INFO[-1] and store other compile model's CPU info in each CPU_INFO[executor_id]. 


> @a3213105 Thanks for your contribution.
> 
> This feature include two part. One is user interface part include new property. The other is feature execution part in CPU plugin and interface folders.
> 
> From design perspective for feature execution part, we just need feature to update cpu_mapping_table and proc_type_table based on input processors before get_num_streams(). Then we do not need much API change with "executor_id". Could you please try it?

**Pull request closed without merging**